src/core/lua/luabfb.cpp

```diff
@@ -57,7 +57,7 @@ const char CLuaBFB::LUA_NAME[] = "FORTE_CLuaFB";
 const luaL_Reg CLuaBFB::LUA_FUNCS[] = { { "__index", CLuaFB_index }, { "__newindex", CLuaFB_newindex }, { "__call", CLuaFB_call }, { nullptr, nullptr } };
 
 CLuaBFB::CLuaBFB(CStringDictionary::TStringId paInstanceNameId, const CLuaBFBTypeEntry* paTypeEntry, forte::core::CFBContainer &paContainer) :
-    CGenFunctionBlock<CBasicFB>(paContainer, paTypeEntry->getInterfaceSpec(), paInstanceNameId, paTypeEntry->getInternalVarsInformation()),
+    CGenFunctionBlock<CBasicFB>(paContainer, paInstanceNameId, paTypeEntry->getInternalVarsInformation()),
```

Comment:
I would suggest to change the return type of `CLuaBFBTypeEntry::getInterfaceSpec()` to a reference, keep passing it to the super constructor here and remove the assignment in `CLuaBFB::createInterfaceSpec()` instead.

Reply:
done. but... what is the purpose of createInterfaceSpec ? It does not seem to get called anywhere

Reply:
The purpose of `createInterfaceSpec(...)` is for blocks where the interface is constructed dynamically based on the `paConfigString` parameter. In case of LUA blocks, the interface is already created in the respective type entry, so we can pass it directly to the super constructor. In that case, `createInterfaceSpec(...)` should simply return true and do nothing otherwise.

Reply:
Ok, done